File: dataminr.js
let buildingHeights = (buildings) => {
  let arr = new Array(buildings.length).fill(0)

  for (let i = 1; i < buildings.length; i++) {
    arr[i] = Math.min(arr[i-1] + 1, buildings[i], i)
  }

  for (let i = arr.length - 1; i > 0; i--) {
    if (arr[i-1] > arr[i] + 1) {
      arr[i-1] = arr[i] + 1
    }
  }

  console.log(arr)
}



buildingHeights([10,1,10,10,10,10,1,10,10])

File: dataminr.py
def buildingHeights(N, buildings, heights):
  d = {buildings[i] - 1: heights[i] for i in range(len(buildings))}
  lst = [0] * N
  for i in range(1, N):
      if i in d and d[i] <= lst[i-1]:
          lst[i] = d[i]
      else:
          lst[i] = lst[i-1] + 1

  print(lst)

  for i in range(N - 2, -1, -1):
      if lst[i] > lst[i + 1] + 1:
          lst[i] = lst[i + 1] + 1

  print(lst)
  return max(lst)

buildingHeights(10, [3,8], [1,1])